azext_edge/edge/providers/orchestration/resources/instances.py

```diff
@@ -391,11 +392,25 @@ def disable_secretsync(
                 return
         logger.warning(f"No secret provider class detected.\n{get_enable_syntax(name, resource_group_name)}")
 
-    def _find_existing_resources(self, cl_resources: List[dict], resource_type: str) -> Optional[List[dict]]:
+    def find_existing_resources(
+        self,
+        cl_resources: List[dict],
+        resource_type: str,
+        resource_name: Optional[str] = None,
+    ) -> Optional[List[dict]]:
         resources = []
+        if not cl_resources:
+            raise ResourceNotFoundError("No resources found in the connected cluster.")
```

Comment:
This is a little misleading, we should say something like:

"No resources found associated with the IoT Operations deployment".

The verbose explanation:

"No custom location resources found associated with the IoT Operations deployment".